### PR TITLE
[RU] Added 'basics/testing' lesson translation.

### PR DESCRIPTION
**Pull request by nscyclone**

### CODE REVIEW

File: ru/lessons/basics/testing.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+layout: page
+title: Тестирование
+category: basics
+order: 12
+lang: ru
+---
+
+Тестирование &mdash; важная часть разработки.  В этом уроке мы узнаем, как тестировать наш код Elixir с помощью ExUnit, и познакомимся с некоторыми отличными приёмами.
```
Comment (brain-geek) [inline ru/lessons/basics/testing.md]: код Elixir -> Elixir код ?

File: ru/lessons/basics/testing.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+- [ExUnit](#section-1)
+  - [assert](#section-2)
+  - [refute](#section-3)
+  - [assert_raise](#section-4)
+- [Настройка теста](#section-5)
+- [Использование "заглушек"](#section-6)
+
+## ExUnit
+
+В Elixir есть встроенный фреймворк для тестирования &mdash; ExUnit и он содержит всё необходимое для тщательного тестирования нашего кода.  Перед тем, как двигаться дальше, важно отметить, что тесты реализованы в виде скриптов Elixir, поэтому нам надо использовать расширение  `.exs`.  Для того, чтобы мы могли выполнять наши тесты, нужно запустить ExUnit с помощью `ExUnit.start()`, обычно это делается в `test/test_helper.exs`.
```
Comment (brain-geek) [inline ru/lessons/basics/testing.md]: I think that `Перед тем, как` is more correct without comma.

File: ru/lessons/basics/testing.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+```shell
+Finished in 0.03 seconds (0.02s on load, 0.01s on tests)
+1 tests, 0 failures
+```
+
+### assert
+
+Если вы когда-либо ранее писали тесты, вы должны быть знакомы с `assert`; в некоторых фреймворках роль `assert` выполняют `should` или `expect`.
+
+Макрос `assert` используется, чтобы проверить, что выражение истинно.  В случае, если это не так, возникнет ошибка, а наши тесты не будут пройдены.  Давайте изменим наш пример и запустим `mix test`, чтобы протестировать ошибку:
```
Comment (brain-geek) [inline ru/lessons/basics/testing.md]: `наши тесты не будут пройдены` -> `тесты завершатся с ошибкой`?

File: ru/lessons/basics/testing.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+     stacktrace:
+       test/example_test.exs:6
+
+......
+
+Finished in 0.03 seconds (0.02s on load, 0.01s on tests)
+1 tests, 1 failures
+```
+
+ExUnit расскажет нам, какое именно утверждение было ошибочным, какое значение ожидалось и какое было получено на самом деле.
```
Comment (brain-geek) [inline ru/lessons/basics/testing.md]: `расскажет нам` -> `покажет`?

File: ru/lessons/basics/testing.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+  test "the truth", state do
+    assert 1 + 1 == state[:number]
+  end
+end
+```
+
+## Использование "заглушек"
+
+Простой ответ касательно использования заглушек в Elixir: не делайте этого.  Возможно, Вам по привычке захочется воспользоваться "заглушкой", но это крайне не приветствуется сообществом Elixir по веским причинам.  Если Вы будете следовать принципам правильного проектирования, конечный код будет легко тестировать в качестве индивидуальных компонентов.
```
Comment (brain-geek) [inline ru/lessons/basics/testing.md]: `Простой ответ` -> `Простой совет`? It would be not very literal translation, but sounds much better.

**Comment by brain-geek** [comment]
@nscyclone , I've reviewed the lesson.

By the way, we have #220 issue to make sure nobody translates a lesson twice.

**Comment by nscyclone** [comment]
Thanks, @brain-geek !
I've just corrected everything according to your last review.

File: ru/lessons/basics/testing.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  - [refute](#section-3)
+  - [assert_raise](#section-4)
+- [Настройка теста](#section-5)
+- [Использование "заглушек"](#section-6)
+
+## ExUnit
+
+В Elixir есть встроенный фреймворк для тестирования &mdash; ExUnit и он содержит всё необходимое для тщательного тестирования нашего кода.  Перед тем как двигаться дальше, важно отметить, что тесты реализованы в виде скриптов Elixir, поэтому нам надо использовать расширение  `.exs`.  Для того, чтобы мы могли выполнять наши тесты, нужно запустить ExUnit с помощью `ExUnit.start()`, обычно это делается в `test/test_helper.exs`.
+
+Когда мы сгенерировали проект-пример из прошлого урока, mix любезно сделал для нас тест, который можно найти в `test/example_test.exs`:
```
Comment (brain-geek) [inline ru/lessons/basics/testing.md]: "любезно" sounds a lot like americanism :)

File: ru/lessons/basics/testing.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+  use ExUnit.Case
+  doctest Example
+
+  test "the truth" do
+    assert 1 + 1 == 2
+  end
+end
+```
+
+Можно выполнить тесты нашего проекта с помощью `mix test`.  Если мы сделаем это, то увидим примерно следующее:
```
Comment (brain-geek) [inline ru/lessons/basics/testing.md]: `Можно выполнить тесты нашего проекта` -> Тесты проекта можно запустить` ?

**Comment by brain-geek** [comment]
@nscyclone I've found several small enhanceable parts, otherwise it's OK.

I'll merge the PR after your response about them.

**Comment by nscyclone** [comment]
I guess it's ready for merge now, @brain-geek .

**Comment by doomspork** [comment]
@brain-geek can I merge this?  Thanks

File: ru/lessons/basics/testing.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+category: basics
+order: 12
+lang: ru
+---
+
+Тестирование &mdash; важная часть разработки.  В этом уроке мы узнаем, как тестировать наш Elixir код с помощью ExUnit, и познакомимся с некоторыми отличными приёмами.
+
+## Содержание
+
+- [ExUnit](#section-1)
```
Comment (brain-geek) [inline ru/lessons/basics/testing.md]: most likely this nav won't work (it's calculated based on english letters)

**Comment by brain-geek** [comment]
Otherwise everything else is OK to merge.

**Comment by nscyclone** [comment]
Thanks for the hint, @brain-geek . I've just fixed and tested the anchors.

**Comment by brain-geek** [comment]
Thanks, @nscyclone !

Sorry for the delay - I've been busy with the new job.

**Comment by doomspork** [comment]
Congrats on the new job @brain-geek :tada:

**Pull request merged**